lisa/operating_system.py

```diff
@@ -1520,9 +1527,14 @@ def add_repository(
 
     def _initialize_package_installation(self) -> None:
         self.wait_running_process("zypper")
-        self._node.execute(
+        output = self._node.execute(
             "zypper --non-interactive --gpg-auto-import-keys refresh", sudo=True
-        )
+        ).stdout
+        if self._no_repo_defined.search(output):
+            raise LisaException(
```

Comment:
Why not use the RepoNotExistException here?

Reply:
I thought there are different issues.
For ubuntu one, the repo definition is there, but repo server is not there.
For sles one, it is either a byos image or encounter repo register issue, there is not any repo listed out if run `zypper lr`